UnityProject/Assets/Scripts/Doors/DoorController.cs

```diff
@@ -1,222 +1,259 @@
+		[Tooltip("Does the denied light animation only toggle 1 frame on and?")] public bool useSimpleDeniedAnimation = false;
+		public float maxTimeOpen = 5;
+		private int openLayer;
+		public AudioSource openSFX;
+		private int openSortingLayer;
+
+		public OpeningDirection openingDirection;
+		private RegisterDoor registerTile;
+		private Matrix matrix => registerTile.Matrix;
+
+		private AccessRestrictions accessRestrictions;
+		public AccessRestrictions AccessRestrictions {
+			get {
+				if ( !accessRestrictions ) {
+					accessRestrictions = GetComponent<AccessRestrictions>();
+				}
+				return accessRestrictions;
+			}
+		}
+
+		[HideInInspector] public SpriteRenderer spriteRenderer;
+
+		public override void OnStartClient()
+		{
+			base.OnStartClient();
+			if (!isWindowedDoor)
+			{
+				closedLayer = LayerMask.NameToLayer("Door Closed");
+			}
+			else
+			{
+				closedLayer = LayerMask.NameToLayer("Windows");
+			}
+			spriteRenderer = GetComponentInChildren<SpriteRenderer>();
+			closedSortingLayer = SortingLayer.NameToID("Doors Closed");
+			openSortingLayer = SortingLayer.NameToID("Doors Open");
+			openLayer = LayerMask.NameToLayer("Door Open");
+
+
+			registerTile = gameObject.GetComponent<RegisterDoor>();
+		}
+
+		/// <summary>
+		/// Invoked by doorAnimator once a door animation finishes
+		/// </summary>
+		public void OnAnimationFinished()
+		{
+			isPerformingAction = false;
+			//check if the door is closing on something, and reopen it if so.
+
+			//When the door first closes, it checks if anything is blocking it, but it is still possible
+			//for a laggy client to go into the door while it is closing. There are 2 cases:
+			// 1. Client enters door after server knows the door is impassable, but before client knows it is impassable.
+			// 2. Client enters door after the close begins but before server marks the door as impassable and before
+			// 		the client knows it is impassable. This is rare but there is a slight delay (.15 s) between when the door close
+			//		begins and when the server registers the door as impassable, so it is possible (See AirLockAnimator.MakeSolid)
+			// Case 1 is handled by our rollback code - the client will be lerp'd back to their previous position.
+			// Case 2 won't be handled by the rollback code because the client enters the passable tile while the
+			//	server still thinks its passable. So, for the rare situation that case 2 occurs, we will apply
+			// the below logic and reopen the door if the client got stuck in the door in the .15 s gap.
+
+			//only do this check when door is closing, and only for doors that block all directions (like airlocks)
+			if (!IsOpened && !registerTile.OneDirectionRestricted)
+			{
+
+				//Check if there's anything occupying this space that's impassable (other than this door)?
+				var registeredTiles = MatrixManager.GetAt<RegisterTile>(transform.position.RoundToInt());
```

Comment:
any reason why you didn't use 
`MatrixManager.IsPassableAt(registerTile.WorldPosition, registerTile.WorldPosition, true, this.gameObject)` here?

Reply:
Yes. The first time I tried fixing it, I DID use that method, but it doesn't work. 

The door itself is already marked as impassable, so this method always returns false. Due to the way passability is determined, it was not feasible to add the ability to exclude the door from the passability check. If you dig into it, passability is checked using TileUtils.IsPassable, which checks against BasicTile - BasicTile does not check passability for the individual objects on it - it's an all or nothing thing.

Reply:
`Layer.IsPassableAt()` overridden in ObjectLayer, and airlock is an Object, check it out:
https://github.com/unitystation/unitystation/blob/0fc17b6faa8a885c292b2d468c40e2fd4144643d/UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/ObjectLayer.cs#L68

Reply:
nice find, thanks for the tip. I'll try that